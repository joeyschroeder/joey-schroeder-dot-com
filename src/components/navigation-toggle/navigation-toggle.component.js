import PropTypes from 'prop-types';
import React from 'react';
import styles from './navigation-toggle.style.scss';

export function NavigationToggle(props) {
  const { onClick } = props;

  return (
    <button className={styles.root} onClick={onClick} type="button">
      <span />
      <span />
      <span />
      <span />
      <span />
      <span />
    </button>
  );
}

NavigationToggle.propTypes = {
  onClick: PropTypes.func,
};

NavigationToggle.defaultProps = {
  onClick: null,
};
