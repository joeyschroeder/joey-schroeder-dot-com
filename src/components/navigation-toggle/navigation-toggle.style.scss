@use "sass:math";

@import "../../styles/config/variables";
@import "../../styles/mixins/transition";

@keyframes navigationTogglePulse {
  0%,
  30%,
  100% {
    transform: scale(0.7);
  }

  15% {
    transform: scale(1.3);
  }
}

.root {
  @include transition(opacity);

  display: flex;
  flex-flow: row wrap;
  position: absolute;
  right: $spacer * 1.5;
  top: $spacer * 1.5;
  width: $spacer * 6;
  z-index: 5;

  span {
    align-items: center;
    animation-duration: $duration-base * 12;
    animation-iteration-count: infinite;
    animation-name: navigationTogglePulse;
    animation-timing-function: ease-in-out;
    display: flex;
    height: $spacer * 2;
    justify-content: center;
    position: relative;
    width: $spacer * 2;

    @for $i from 1 through 6 {
      &:nth-child(#{$i}) {
        animation-delay: $i * math.div($duration-base, 3);
      }
    }

    &::after {
      border-radius: 100%;
      content: "";
      height: $spacer;
      position: absolute;
      width: $spacer;
    }

    &:nth-child(1)::after,
    &:nth-child(4)::after {
      background: $color-stripe-3;
    }

    &:nth-child(2)::after,
    &:nth-child(5)::after {
      background: $color-stripe-2;
    }

    &:nth-child(3)::after,
    &:nth-child(6)::after {
      background: $color-stripe-1;
    }
  }
}
