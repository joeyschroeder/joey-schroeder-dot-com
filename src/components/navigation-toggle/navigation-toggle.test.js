import React from 'react';
import { create } from 'react-test-renderer';
import { NavigationToggle } from './navigation-toggle.component';

describe('<NavigationToggle />', () => {
  it('should render correctly', () => {
    const component = create(<NavigationToggle />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
