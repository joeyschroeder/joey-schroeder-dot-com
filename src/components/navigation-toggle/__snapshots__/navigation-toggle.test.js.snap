// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationToggle /> should render correctly 1`] = `
<button
  className="root"
  onClick={null}
  type="button"
>
  <span />
  <span />
  <span />
  <span />
  <span />
  <span />
</button>
`;
